src/main/java/org/broadinstitute/dsm/jobs/PubSubLookUp.java

```diff
@@ -0,0 +1,144 @@
+
+            logger.error("A new result for sample id "+testBostonResult.getSampleId() +" that doesn't match the previous one. Date of the new result: "+testBostonResult.getTimeCompleted());
+            return false;
+        }
+        return true;
+    }
+
+    private static void tellPepperAboutTheNewResults(TestBostonResult testBostonResult) {
+        String query = "select   eve.event_name,   eve.event_type,   request.ddp_participant_id,   request.dsm_kit_request_id,   realm.ddp_instance_id,   realm.instance_name,   realm.base_url,   realm.auth0_token,   realm.notification_recipients,   realm.migrated_ddp,   kit.receive_date,   kit.scan_date   from   ddp_kit_request request,   ddp_kit kit,   event_type eve,   ddp_instance realm   where request.dsm_kit_request_id = kit.dsm_kit_request_id   and request.ddp_instance_id = realm.ddp_instance_id   " +
+                " and (eve.ddp_instance_id = request.ddp_instance_id   and eve.kit_type_id = request.kit_type_id)   and eve.event_type = \"RESULT\" " + // that's the change from the original query
+                " and request.external_order_number = ?"; // that's the change from the original query
+
+        KitDDPNotification kitDDPNotification = KitDDPNotification.getKitDDPNotification(query, testBostonResult.getSampleId());
+        if (kitDDPNotification != null) {
+            EventUtil.triggerDDPWithTestResult(kitDDPNotification, testBostonResult);
+        }
+    }
+
+    private static DSMTestResult[] getLatestKitTestResults(TestBostonResult testBostonResult) {
+        String externalOrderNumber = testBostonResult.getSampleId();
+        SimpleResult results = inTransaction((conn) -> {
+            SimpleResult dbVals = new SimpleResult();
+            try (PreparedStatement stmt = conn.prepareStatement(SELECT_LATEST_RESULT_QUERY)) {
+                stmt.setString(1, externalOrderNumber);
+                try (ResultSet rs = stmt.executeQuery()) {
+                    if (rs.next()) {
+                        String resultArray = rs.getString("test_result");
+                        DSMTestResult[] dsmTestResultsArray = new Gson().fromJson(resultArray, DSMTestResult[].class);
+                        dbVals.resultValue = dsmTestResultsArray;
+                    }
+                }
+                catch (SQLException ex) {
+                    dbVals.resultException = ex;
+                }
+            }
+            catch (SQLException ex) {
+                dbVals.resultException = ex;
+            }
+            return dbVals;
+        });
+
+        if (results.resultException != null) {
+            throw new RuntimeException("", results.resultException);
+        }
+        DSMTestResult[] dsmTestResult = (DSMTestResult[]) results.resultValue;
+        return dsmTestResult;
+    }
+
+    public static void writeResultsIntoDB(TestBostonResult testBostonResult) {
+        DSMTestResult[] dsmTestResultArray = getLatestKitTestResults(testBostonResult);
```

Comment:
One optimization is to just do one single query for latest kit results. We can do the query in the parent process method and pass that array into the `shouldWriteResultIntoDB()` and this method.